dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dataverse/DataversePage.java

```diff
@@ -116,27 +124,21 @@ public boolean isRootDataverse() {
 
     // -------------------- LOGIC --------------------
     public String init() {
-
-        if (dataverse.getAlias() != null) {
-            dataverse = dataverseDao.findByAlias(dataverse.getAlias());
-        } else if (dataverse.getId() != null) {
-            dataverse = dataverseDao.find(dataverse.getId());
+        if (StringUtils.isNotEmpty(dataverseAlias)) {
+            dataverse = dataverseDao.findByAlias(dataverseAlias);
+        } else if(dataverseId != null) {
+            dataverse = dataverseDao.find(dataverseId);
         } else {
-            try {
-                dataverse = dataverseDao.findRootDataverse();
-            } catch (EJBException e) {
-                // @todo handle case with no root dataverse (a fresh installation) with message about using API to create the root
-                dataverse = null;
-            }
+            dataverse = dataverseDao.findRootDataverse();
```

Comment:
what about possible exceptions from `findRootDataverse()`?

Reply:
Not sure if this was really needed but I'll make sure we catch failure there.